### ISSUE TITLE
Despite synchronising playback to the server, songs are not scrobbled to last.fm

### ISSUE THREAD

**Issue by el-tiuri** [UnicornsOnLSD/finamp#87]
Now, I have no clue if the issue here is with FinAmp or the last.fm plugin. Either way, as the title says, in the admin dashboard I can verify that it is reporting my account playing the song I’m actually playing. However, no music is ever scrobbled despite the last.fm being configured for my account.

It just works with the main Jellyfin app, so I’m wondering if FinAmp might report the tracks being played in a slightly different way or something.

**Comment by UnicornsOnLSD** [comment]
Finamp probably isn't doing the full track reporting properly, I'll look into it.

**Comment by UnicornsOnLSD** [comment]
It was working fine for me, does last.fm work when you play songs via the Jellyfin web interface?

**Comment by vdatanet** [comment]
Yes, when I play using Jellyfin web interface, it scrobbles to last.fm. FinAmp it doesn't, but used to work. I'd swear it worked last week.

**Comment by el-tiuri** [comment]
Same for me. Scrobbling works when using the web interface or the Jellyfin app, but not when using FinAmp.

**Comment by vdatanet** [comment]
Even the songs don’t appear as recently played on Jellyfin’s music recommendations.

**Comment by deafmute1** [comment]
+1 on this, neither the listenbrainz nor the lastfm scrobbler plugins work for me via finamp. Scrobbling  does work for the following clients for me: 
- mopidy
- web UI 
- official desktop client.

Just started using finamp, so can't speak to it being new or old issue.

**Comment by gerardlouw** [comment]
In https://github.com/jesseward/jellyfin-plugin-lastfm/issues/27#issuecomment-744031810 it is mentioned that the plugin expects a `PlaybackStartEvent` and `PlaybackStopEvent` pair to be received by the Jellyfin server for it to register a scrobble. I did a quick code search here and that not find any direct references to these, so I believe that may be what is required for a fix.

**Comment by UnicornsOnLSD** [comment]
Should be fixed in 7d56bdd, I'll close this when someone else can verify that it works

**Comment by vdatanet** [comment]
Hi,

In 0.5.1 scrobbling to last.fm is fixed and played songs are correctly registered in recently played songs in Jellyfin. Thanks!

**Comment by gerardlouw** [comment]
For me on 0.5.1 it is correctly showing the "start" and "finish" events in the Jellyfin activity log. On last.fm it shows that I am busy listening to the song while it is playing, but once it finishes it the scrobble is not actually recorded. Maybe this is a problem not with Finamp.

**Comment by el-tiuri** [comment]
Works perfectly for me now!

**Comment by el-tiuri** [comment]
It works for me, but seemingly only with music I haven’t downloaded to my phone.

**Comment by vdatanet** [comment]
Last track of a queue is scrobbled twice:

![Captura de pantalla 2021-08-08 122401](https://user-images.githubusercontent.com/22084256/128628843-9bb3a816-a7e2-4556-abab-b9ba5bf6d0fb.png)
![Captura de pantalla 2021-08-08 122331](https://user-images.githubusercontent.com/22084256/128628845-b1922e95-6a32-4594-8bfe-717de950b0bd.png)

**Comment by UnicornsOnLSD** [comment]
> It works for me, but seemingly only with music I haven’t downloaded to my phone.

It won't scrobble in offline mode, but it should still scrobble downloaded songs when not in offline mode.

**Comment by UnicornsOnLSD** [comment]
Double scrobbling is hopefully fixed in 98da06d, I'll test it properly next time I use the app

**Comment by el-tiuri** [comment]
> > It works for me, but seemingly only with music I haven’t downloaded to my phone.
> 
> It won't scrobble in offline mode, but it should still scrobble downloaded songs when not in offline mode.

Weird, I’d swear it didn’t work before, but it does now. That’s fantastic!

Is syncing the playback state when you take your device back online planned, by the way? I’d love for scrobbling to work in offline mode.

**Comment by carks32** [comment]
maybe it's a me problem since this got fixed for so many people but it seems that starting a playlist/album just doesn't work to scrobble, they appear just fine on the activity tab and they do appear as scrobbling on last.fm but they never end up getting scrobbled.
![image](https://user-images.githubusercontent.com/14918982/130339642-b55de9a2-bfaa-4f33-aec6-30ba513bb8ba.png)

**Comment by UnicornsOnLSD** [comment]
> Is syncing the playback state when you take your device back online planned, by the way? I’d love for scrobbling to work in offline mode.

If Jellyfin has a proper way to send bulk playback data, I'll add support. I don't want to just spam start/end playbacks though

**Comment by UnicornsOnLSD** [comment]
> maybe it's a me problem since this got fixed for so many people but it seems that starting a playlist/album just doesn't work to scrobble, they appear just fine on the activity tab and they do appear as scrobbling on last.fm but they never end up getting scrobbled.
> ![image](https://user-images.githubusercontent.com/14918982/130339642-b55de9a2-bfaa-4f33-aec6-30ba513bb8ba.png)

I think last.fm lags behind a bit, could you check after a few hours?

**Comment by carks32** [comment]
> I think last.fm lags behind a bit, could you check after a few hours?

after 16 hours, it seems that is just not the case. How could I try to get you logs while I try to replicate this?

**Comment by UnicornsOnLSD** [comment]
The music I listened to last night wasn't synced, so it might actually still be an issue. I'll look into it more

**Comment by UnicornsOnLSD** [comment]
Progress updates are slightly different in the branch I'm working on right now, so there isn't much point in sending logs at the moment.

**Comment by icyphox** [comment]
hey, apologies for bumping an old issue, but i’m facing this exact bug on iOS 15, downloaded from the App Store.

**Comment by UnicornsOnLSD** [comment]
I still haven't released those improvements from August, new release soon™ (hopefully before the end of November)

**Comment by Maxr1998** [comment]
I'm still using the build from last August, but I think I know why it doesn't submit even if both start and stop events register correctly.

The ListenBrainz plugin (and likely Last.fm too) only submits listens [if a (configurable) minimum percentage of the track was played](https://github.com/lyarenei/jellyfin-plugin-listenbrainz/blob/master/Jellyfin.Plugin.Listenbrainz/ServerEntryPoint.cs#L90), this requires the stop event to contain the correct duration value. According to the logs on my server, the value is way too small, causing the listen to get dropped.

```
Listen to '…' won't be submitted, played 0.0008246693926028897% (15770 ticks), required 50% or 2400000000 ticks
```

It seems that it uses the progress for the following track in the stop event, which is still at the start.

**Comment by powellc** [comment]
Any chance this is gonna get merged soon? I love Finamp and use it all the time. But it's killing me that all my Finamp listening isn't being scrobbled properly :D

**Comment by Chaphasilor** [comment]
@Maxr1998 initially fixed this in #200, and #484 should adapt his changes into the new queue system...

**Comment by Chaphasilor** [comment]
Playback reporting and scrobbling has been improved as part of the redesign beta. You can check it out here: https://github.com/jmshrv/finamp/releases/tag/0.9.2-beta

I'll close this issue for now since it should be resolved.

**Comment by JaggedJax** [comment]
I am seeing this same issue when playing using the Flatpak desktop app (0.9.12).

It is working fine using the Android app (0.9.13)

Edit: I'm wondering if this is just because the Flatpack doesn't have 0.9.13 available, and 0.9.13 has this change listed: `Fixed reported playback not properly stopping (or reappearing later) on the server`

Edit Edit: Confirming this is working for me on Flatpak now
